docs/statistics/hypothesis/Testing.md

```diff
@@ -0,0 +1,189 @@
+
+> In statistical hypothesis testing, the null distribution is the probability distribution of the test statistic when the null hypothesis is true.
+>
+> -- <cite>Kent W. Staley - An Introduction to the Philosophy of Science</cite>
+
+To build the null distribution:
+
+1. **Repeat Sampling**: Take many pairs of samples from the population.
+2. **Calculate Differences**: For each pair, compute the difference in sample means.
+3. **Plot the Differences**: Create a histogram of these differences.
+
+So if we stick with our `age` example, and compare samples from the whole population, we will receive a distribution of age differences:
+
+<iframe src="/assets/statistics/prob_testing_nulldist.html" width="100%" height="400px"></iframe>
+
+??? code "Code"
+    ``` py
+    from ucimlrepo import fetch_ucirepo 
+    
+    # fetch dataset 
+    adult = fetch_ucirepo(id=2) 
+    
+    # data (as pandas dataframes) 
+    data = adult.data.features
+
+    import plotly.express as px
+    import random
+    import numpy as np
+    import pandas as pd
+
+    sample_means_dif = []
+    for i in range(10000):
+        sample_means_dif.append(np.mean(random.sample(list(data.age), 20))-np.mean(random.sample(list(data.age), 20)))
+
+
+    fig = px.histogram(x=sample_means_dif)
+    fig.update_traces(marker=dict(color='rgba(0, 65, 110, 0.8)'))
+
+    # Add a horizontal line for the population mean
+    fig.add_vline(x=0, line_dash="dash", annotation_text="Expected Value: 0", annotation_position="top right", line_color="#E87F2B")
+
+    fig.update_layout(
+            barmode='overlay',
+            yaxis_title_text='Absolute Frequency',
+            xaxis_title_text='Sample Means Difference',
+            title=dict(
+                text='<b><span style="font-size: 10pt">Null Distribution</span><br><span style="font-size: 8pt">Age vs. Age</span><span style="font-size: 5pt"> Number of Samples: 10.000 | Data: UCIML Repo: Adult; variable: age</span> </b>',
+            ),
+        )
+
+    fig.show()
+
+    ```
+In this case we see, that the distribution looks like we would imagine if the null hypothesis cannot be rejected (is true). 
+
+### Alternative Distribution
+
+The alternative hypothesis posits that there is a genuine difference between groups. Therefore, under the alternative hypothesis, the distribution of differences shifts away from zero. The **alternative distribution** represents the expected differences if the alternative hypothesis is true.
+
+Again we stick with the `age` example and compare the age differences of mutliple samples for `male` and `female` in our dataset:
+
+
+<iframe src="/assets/statistics/prob_testing_altdist.html" width="100%" height="400px"></iframe>
+
+??? code "Code"
```

Comment:
Do you still have the code snippet for the alternative distribution? If I'm not mistaken, that's the same example as in the null distribution. 🙈

Reply:
I think its the right one. @ the alternative distribution we have an aditional condition data['sex']=="Male/Female"